pytorch_lightning/trainer/trainer.py

```diff
@@ -2074,7 +2074,7 @@ def model(self, model: torch.nn.Module) -> None:
             model: The LightningModule, possibly wrapped into DataParallel or DistributedDataParallel, depending
                 on the backend.
         """
-        self.strategy.model = model
+        self.strategy.connect(model)
```

Comment:
I believe I can undo this change, this would only impact TPUs

```suggestion
        self.strategy.model = model
```

But I wonder if calling connect is the best solution long-term as that's where Strategy authors can add their logic.